src/utils/metadata_integration.py

```
@@ -0,0 +1,579 @@
+                "frontmatter": structured_metadata["frontmatter"],
+                "frontmatter_validation": structured_metadata["frontmatter_validation"],
+                # Additional metadata
+                "source_context": source_context or {},
+                "quality_metrics": _calculate_quality_metrics(
+                    structured_metadata, keywords_with_scores, technical_entities
+                ),
+            }
+
+            return complete_metadata
+
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)